src/solver/fold/fold_forward.cpp

```diff
@@ -0,0 +1,167 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "miopen/fold/problem_description.hpp"
+#include "miopen/miopen.h"
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/fold/invoke_params.hpp>
+#include <miopen/fold/solvers.hpp>
+#include <miopen/fold.hpp>
+#include <miopen/target_properties.hpp>
+#include <miopen/tensor_view_utils.hpp>
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace fold {
+
+bool FoldFwd::IsApplicable([[maybe_unused]] const ExecutionContext& /*context*/,
+                           const miopen::fold::FoldFwdProblemDescription& problem) const
+{
+    return true;
+}
+
+ConvSolution FoldFwd::GetSolution([[maybe_unused]] const ExecutionContext& context,
+                                  const miopen::fold::FoldFwdProblemDescription& problem) const
+{
+    std::ignore = context;
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto in_dtype   = miopen::GetDataType(problem.GetInputDesc().GetType());
+    auto dtype      = problem.GetOutputDesc().GetType();
+    auto input_dims = problem.GetInputDesc().GetLengths();
+
+    auto output_dims = problem.GetOutputDesc().GetLengths();
+    const int32_t N  = static_cast<int32_t>(output_dims[0]);
+    const int32_t C  = static_cast<int32_t>(output_dims[1]);
+    int32_t H        = static_cast<int32_t>(output_dims[2]);
+    int32_t W        = static_cast<int32_t>(output_dims[3]);
+
+    {
+        auto kernel        = KernelInfo{};
+        kernel.kernel_file = "MIOpenUnfold.cpp";
+        kernel.kernel_name = "UnfoldBackward4D";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"IN_OUT_TYPE", in_dtype == "bfloat16" ? "ushort" : in_dtype},
```

Comment:
Similarly, don't use this if you don't have a need for it. `FLOAT` works fine

Reply:
resolved in [bd5db59](https://github.com/moreh-dev/MIOpen/pull/37/commits/bd5db593af74bd68f03e44465a4a1beec55b8056)